pennylane_lightning/core/src/measurements/MeasurementsBase.hpp

```diff
@@ -111,6 +114,183 @@ template <class StateVectorT, class Derived> class MeasurementsBase {
     auto generate_samples(size_t num_samples) -> std::vector<size_t> {
         return static_cast<Derived *>(this)->generate_samples(num_samples);
     };
-};
 
-} // namespace Pennylane::Measures
\ No newline at end of file
+    /**
+     * @brief Calculate the expectation value for a general Observable.
+     *
+     * @param obs Observable.
+     * @param num_shots Number of shots used to generate samples
+     * @param shot_range The range of samples to use. All samples are used
+     * by default.
+     *
+     * @return Expectation value with respect to the given observable.
+     */
+    auto expval(const Observable<StateVectorT> &obs, const size_t &num_shots,
+                const std::vector<size_t> &shot_range = {}) -> PrecisionT {
+        PrecisionT result = 0;
+
+        if (obs.getObsName().find("SparseHamiltonian") != std::string::npos) {
+            PL_ABORT("For SparseHamiltonian Observables, expval calculation is "
```

Comment:
Let's add some code comments that why this support doesn't make sense for SparseHamiltonian for the future reference.

Reply:
Done!